webcompat/static/js/lib/issues.js

```diff
@@ -547,8 +547,10 @@ issues.MainView = Backbone.View.extend(
       var target =
         e.target.nodeName === "IMG"
           ? e.target
-          : e.target.nodeName === "P" && e.target.firstElementChild;
+          : e.target.nodeName === "P" && e.target.getElementsByTagName("img");
```

Comment:
Can you help me understand this change?

Reply:
Yes!
When I fetch new images from github using this html format githubs gives me a response that contains an image nested on a link, like this:

```
<p class="issue-details-nsfw">
    <a target="_blank" rel="noopener noreferrer" href="https://cloud.githubusercontent.com/assets/67283/15969039/bd82fae2-2ef3-11e6-800f-535785681b50.jpg">
        <img src="https://cloud.githubusercontent.com/assets/67283/15969039/bd82fae2-2ef3-11e6-800f-535785681b50.jpg" alt="29upbiv" style="max-width:100%;">. 
    </a>
</p>
```

And the original code was expecting something like this:

```
<p class="issue-details-nsfw">
    <img src="https://cloud.githubusercontent.com/assets/67283/15969039/bd82fae2-2ef3-11e6-800f-535785681b50.jpg" alt="29upbiv" style="max-width:100%;">. 
</p>
```

And every time I clicked on the image instead of only toggling the the blur on/off it was also opening the image in a new tab because of the <a>. So... I though I could just deactivate it, since Github fetches this automatically. You think it's bad implementation choice to do that way? And have a link that doesn't go everywhere? 
I could try to remove it when the issue is being fetched, so it wouldn't even be part of the code... Even though I don't know how I'd do that. Maybe with regex?

Reply:
Hm. I'm not sure about what the right thing to do is here... I'll leave that up to you. :)

But we should use `querySelector` instead of `getElementsByTagName` for the reasons mentioned above.

And [using regex to parse & manipulate HTML is super painful](https://blog.codinghorror.com/parsing-html-the-cthulhu-way/). If we have to go that way, it would make more sense to manipulate the DOM when we render the body/comment, the way you already are doing.

It's maybe possible to just prevent the default action on that `<a>` and dispatch a click on the `<img`>, but I'm not sure it's worth the extra effort for such a corner case feature.